cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/MetricSample.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.monitor.sampling;
+
+import com.linkedin.cruisecontrol.resource.Resource;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MetricSample<E> {
```

Comment:
It's not clear what type parameter E is supposed to represent.  @type javadoc can be used to provide documentation.  So is en Entity equivelent to a broker?

Reply:
Also E is used as a key in a map so it needs to implement hashCode() and equals().  This needs to be documented somewhere.

Reply:
Also I really want the type parameter to be E extends Comparable<E> so I can eventually fulfill the dream of having a deterministic cruise control.